modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java

```diff
@@ -1584,6 +1587,121 @@ public void testGetWithSystemObjectFields() throws Exception {
+		String dateTimePropertyValue = _dateTimeFormat.format(
+			RandomTestUtil.nextDate());
+		double decimalPropertyValue = RandomTestUtil.randomDouble();
+		int integerPropertyValue = RandomTestUtil.randomInt();
+		long longIntegerPropertyValue = RandomTestUtil.randomLong(
+			ObjectFieldValidationConstants.BUSINESS_TYPE_LONG_VALUE_MIN,
+			ObjectFieldValidationConstants.BUSINESS_TYPE_LONG_VALUE_MAX);
+		String longTextPropertyValue = RandomTestUtil.randomString();
+		Serializable multiselectPicklistPropertyValue =
+			(Serializable)TransformUtil.transform(
+				Arrays.asList(ListTypeValue.VALUE1, ListTypeValue.VALUE3),
+				ListTypeValue::name);
+		String picklistPropertyValue = ListTypeValue.VALUE1.name();
+		double precisionDecimalPropertyValue = RandomTestUtil.randomDouble();
+		String richTextPropertyValue = RandomTestUtil.randomString();
+		String textPropertyValue = RandomTestUtil.randomString();
+		String textUniquePropertyValue = "Unique field value";
+
+		JSONObject jsonObject = HTTPTestUtil.invokeToJSONObject(
+			JSONUtil.put(
+				"attachmentProperty", attachmentPropertyIdValue
+			).put(
+				"booleanProperty", booleanPropertyValue
+			).put(
+				"dateProperty", datePropertyValue
+			).put(
+				"dateTimeProperty", dateTimePropertyValue
+			).put(
+				"decimalProperty", decimalPropertyValue
+			).put(
+				"integerProperty", integerPropertyValue
+			).put(
+				"longIntegerProperty", longIntegerPropertyValue
+			).put(
+				"longTextProperty", longTextPropertyValue
+			).put(
+				"multiselectPicklistProperty", multiselectPicklistPropertyValue
+			).put(
+				"picklistProperty", picklistPropertyValue
+			).put(
+				"precisionDecimalProperty", precisionDecimalPropertyValue
+			).put(
+				"richTextProperty", richTextPropertyValue
+			).put(
+				"textProperty", textPropertyValue
+			).put(
+				"textUniqueProperty", textUniquePropertyValue
+			).toString(),
+			StringBundler.concat("c/", _BASE_URL_1, "/test"), Http.Method.POST);
+
+		JSONAssert.assertEquals(
+			JSONUtil.put(
+				"booleanProperty", booleanPropertyValue
+			).put(
+				"dateProperty", datePropertyValue
+			).put(
+				"dateTimeProperty", dateTimePropertyValue
+			).put(
+				"integerProperty", integerPropertyValue
+			).put(
+				"longTextProperty", longTextPropertyValue
+			).put(
+				"multiselectPicklistProperty",
+				JSONUtil.putAll(
+					JSONUtil.put(
+						"key", ListTypeValue.VALUE1
+					).put(
+						"name", ListTypeValue.VALUE1
+					),
+					JSONUtil.put(
+						"key", ListTypeValue.VALUE3
+					).put(
+						"name", ListTypeValue.VALUE3
+					))
+			).put(
+				"picklistProperty",
+				JSONUtil.put(
+					"key", ListTypeValue.VALUE1
+				).put(
+					"name", ListTypeValue.VALUE1
+				)
+			).put(
+				"richTextProperty", richTextPropertyValue
+			).put(
+				"textProperty", textPropertyValue
+			).put(
+				"textUniqueProperty", textUniquePropertyValue
+			).toString(),
+			jsonObject.toString(), JSONCompareMode.LENIENT);
+
+		Assert.assertEquals(
```

Comment:
I tried to replace with the following code:

```java
JSONObject jsonObject = HTTPTestUtil.invokeToJSONObject(
	JSONUtil.put(
		"attachmentProperty", attachmentPropertyIdValue
	).put(
		"booleanProperty", booleanPropertyValue
	).put(
		"dateProperty", datePropertyValue
	).put(
		"dateTimeProperty", dateTimePropertyValue
	).put(
		"decimalProperty", decimalPropertyValue
	).put(
		"integerProperty", integerPropertyValue
	).put(
		"longIntegerProperty", (Long)longIntegerPropertyValue
	).put(
		"longTextProperty", longTextPropertyValue
	).put(
		"multiselectPicklistProperty", multiselectPicklistPropertyValue
	).put(
		"picklistProperty", picklistPropertyValue
	).put(
		"precisionDecimalProperty", precisionDecimalPropertyValue
	).put(
		"richTextProperty", richTextPropertyValue
	).put(
		"textProperty", textPropertyValue
	).put(
		"textUniqueProperty", textUniquePropertyValue
	).toString(),
	StringBundler.concat("c/", _BASE_URL_1, "/test"), Http.Method.POST);

JSONAssert.assertEquals(
	JSONUtil.put(
		"attachmentProperty",
		JSONUtil.put("id", attachmentPropertyIdValue)
	).put(
		"booleanProperty", booleanPropertyValue
	).put(
		"dateProperty", datePropertyValue
	).put(
		"dateTimeProperty", dateTimePropertyValue
	).put(
		"decimalProperty", decimalPropertyValue
	).put(
		"integerProperty", integerPropertyValue
	).put(
		"longIntegerProperty", (Long)longIntegerPropertyValue
	).put(
		"longTextProperty", longTextPropertyValue
	).put(
		"multiselectPicklistProperty",
		JSONUtil.putAll(
			JSONUtil.put(
				"key", ListTypeValue.VALUE1
			).put(
				"name", ListTypeValue.VALUE1
			),
			JSONUtil.put(
				"key", ListTypeValue.VALUE3
			).put(
				"name", ListTypeValue.VALUE3
			))
	).put(
		"picklistProperty",
		JSONUtil.put(
			"key", ListTypeValue.VALUE1
		).put(
			"name", ListTypeValue.VALUE1
		)
	).put(
		"precisionDecimalProperty", precisionDecimalPropertyValue
	).put(
		"richTextProperty", richTextPropertyValue
	).put(
		"textProperty", textPropertyValue
	).put(
		"textUniqueProperty", textUniquePropertyValue
	).toString(),
	jsonObject.toString(), JSONCompareMode.LENIENT);
```

And it mainly, works, the only failure is related to the `precisionDecimalProperty`, but the problem is the data we are randomly generating:

```
    java.lang.AssertionError: precisionDecimalProperty
    Expected: 0.23270647620276708
         got: 0.2327064762027671
```

Please notice we are using 17 decimals instead of 16.

See the following constants:

```java
public class ObjectFieldValidationConstants {

	[...]

	public static final String BUSINESS_TYPE_PRECISION_DECIMAL_VALUE_MAX =
		"99999999999999.9999999999999999";

	public static final String BUSINESS_TYPE_PRECISION_DECIMAL_VALUE_MIN =
		"-99999999999999.9999999999999999";

}
```

I guess in this test we should use a double with max 16 decimals, I don't think this is a test where we should manage more than 16 decimals